File: php/testcode/benchmark_test_00375.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00375(BenchmarkRequest $req): BenchmarkResponse {
    $pdo = getDbConnection();
    $id = $req->param('id');
    $stmt = $pdo->prepare("SELECT username FROM users WHERE id = ?");
    $stmt->execute([$id]);
    $username = $stmt->fetchColumn();
    $result = $pdo->query("SELECT * FROM logs WHERE actor = '" . $username . "'");
    $rows = $result->fetchAll();
    return BenchmarkResponse::json($rows);
}

File: ruby/testcode/benchmark_test_00375.rb
require_relative 'shared'

def track_email_open(req)
  email_id = req.param('id').to_i
  db = get_db_connection
  db.execute("UPDATE emails SET opened_at = ? WHERE id = ?", [Time.now.to_s, email_id])
  BenchmarkResponse.json({ result: '' })
end
